test/parallel/test-stdout-close-unref.js

```diff
@@ -1,16 +1,29 @@
 'use strict';
-require('../common');
-var assert = require('assert');
+const common = require('../common');
+const assert = require('assert');
+const spawn = require('child_process').spawn;
 
-var errs = 0;
+if (process.argv[2] === 'child') {
+  var errs = 0;
 
-process.stdin.resume();
-process.stdin._handle.close();
-process.stdin._handle.unref();  // Should not segfault.
-process.stdin.on('error', function(err) {
-  errs++;
-});
+  process.stdin.resume();
+  process.stdin._handle.close();
+  process.stdin._handle.unref();  // Should not segfault.
+  process.stdin.on('error', function(err) {
+    errs++;
+  });
 
-process.on('exit', function() {
-  assert.strictEqual(errs, 1);
-});
+  process.on('exit', function() {
+    assert.strictEqual(errs, 1);
+  });
+  return;
+}
+
+// Use spawn so that we can be sure that stdin has a _handle property.
+// Refs: https://github.com/nodejs/node/pull/5916
+const proc = spawn(process.execPath, [__filename, 'child'], { stdio: 'pipe' });
+
+proc.stderr.pipe(process.stderr);
+proc.on('exit', common.mustCall(function(exitCode) {
+  process.exit(exitCode);
```

Comment:
@Fishrock123 I suppose so. I did that instead of setting `process.exitCode` because setting the exit code but doing nothing else is kind of magical and also leaves open the possibility of someone adding a test to this file later on that does the same thing, thus overriding the exit code here. I'm probably overthinking it, though. Will switch to your recommendation.